src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -718,3 +726,27 @@
              {:current-screen stack-id
               :start-flow?    start-flow?
               :flow-id        flow-id}]]]})))
+
+(rf/reg-event-fx
+ :wallet/transaction-confirmation-navigate-back
+ (fn [{db :db} [{:keys []}]]
+   (let [tx-type       (-> db :wallet :ui :send :tx-type)
+         keep-tx-data? (#{:account-collectible-tab :wallet-stack}
+                        (-> db :wallet :ui :send :entry-point))]
+     {:db (cond-> db
+            (and (= tx-type :tx/collectible-erc-721) (not keep-tx-data?))
+            (update-in [:wallet :ui :send] dissoc :tx-type :amount :route :suggested-routes)
+
+            (= tx-type :tx/collectible-erc-1155)
+            (update-in [:wallet :ui :send] dissoc :route :suggested-routes))
+      :fx [[:dispatch [:navigate-back]]]})))
+
+(rf/reg-event-fx
+ :wallet/collectible-amount-navigate-back
+ (fn [{db :db} [{:keys []}]]
+   (let [keep-tx-data? (#{:account-collectible-tab :wallet-stack}
+                        (-> db :wallet :ui :send :entry-point))]
+     {:db (cond-> db
+            :always             (update-in [:wallet :ui :send] dissoc :amount :route)
+            (not keep-tx-data?) (update-in [:wallet :ui :send] dissoc :tx-type))
```

Comment:
I think this can be:
```
(-> db
              (update-in [:wallet :ui :send] dissoc :amount :route)
              (when (not keep-tx-data?)
                (update-in [:wallet :ui :send] dissoc :tx-type)))
```

Reply:
The `->` is a macro, not a funciton, so the `when` clause won't work as expected @OmarBasem

Reply:
BTW, thanks for the suggestion, actually if `->` were a function, it'd be an interesting way to solve it :+1: